ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -116,12 +107,35 @@ def process_pixels(image_bytes: bytes, expected_channels: int) -> np.ndarray:
             # Didn't find the header, so must be at the end.
             break
 
-    img = np.concatenate(image_arrays, axis=2)
-    # We can drop additional channels since they may need to be added to include
-    # numbers of observation channels not divisible by 3.
-    actual_channels = list(img.shape)[2]
-    if actual_channels > expected_channels:
-        img = img[..., 0:expected_channels]
+    if mappings is not None and len(mappings) > 0:
+        image_arrays = np.concatenate(image_arrays, axis=2).transpose((2, 0, 1))
```

Comment:
Can you do something like
```
mappings = list(range(expected_channels)) if expected_channels > 0 else [0, 0, 0]
```
so that the "old case" is handled by the new code?

Reply:
Doesn't that get into the same performance debate for C# that a mapping has to be created every time?

Reply:
Fair point :) I'm not sure if memory allocation/GC in python is better or worse than in C#, but it's at least tolerated in python.

Reply:
Looks like you could use `range(expected_channels)` (although you'd have to make the mypy types a bit more generic) which should allocate less, and you could reuse `(0, 0, 0)`

Reply:
I've split them into two functions now so it looks clear how it process with mapping or with expected_channels. If unifying them still looks better than this I can change it.